### DISCUSSION TITLE
'analyze learn string'

### DISCUSSION THREAD

**Discussion by Aspire-Design-Ltd** [VK2BEA/HP8753-Companion#15] [General]
Hi Michael. 
Why is the  'analyze learn string'  required?
What extra functionality do you get? The software seems to work fine without using  'analyze learn string'. 
Thanks. MPC.

**Comment by VK2BEA** [comment]
look at the code `HP8753comms.c` where there are references to `pGlobal->HP8753.pLSindexes`
Some of the effects are subtle .. returning to the 'active channel' for example. Also, knowing the 'active channel' permits me to do some actions without having to switch to the channel explicitly which speeds things up under some scenarios (slow sweep). As I said, the knowledge of what markers are on or off (and which is active) is not obtainable with regular HPIB commands. (I think later versions of the firmware (yours for example) may have a command to determine the active channel but mine does not)

```
// Index to HP8753 learn string for items that we cannot
// get with conventional queries.
// This will no doubt be different for every firmware version, so if in doubt
// we don't access markers
tLearnStringIndexes learnStringIndexes[] = {
        {
          .version        = 413,          // version valid for the data below
          .iActiveChannel = 1859,         // active channel (0x01 or 0x02)
          .iMarkersOn     = {2323, 2325}, // markers on     (bit or of 0x02 (marker 1) to 0x10 (marker 4) with 0x20 as all off)
          .iMarkerActive  = {1285, 1378}, // current marker (0x02 (marker 1) to 0x10 (marker 4))
          .iMarkerDelta   = {1286, 1379}, // (bit or of 0x02 (marker 1) to 0x10 (marker 4) with 0x20 (fixed) 0x40 as all off)
          .iStartStop     = {2383, 2385}, // stimulus start/stop or center/span (0x01 is start/stop)
          .iSmithMkrType  = {1289, 1382}, // Smith marker type 0x00 - Lin / 0x01 - Log / 0x02 - Re-Im / 0x04 - R+jX / 0x08 - G+jB
          .iPolarMkrType  = {1288, 1381}, // Polar marker type 0x10 - Lin / 0x20 - log / 0x40 - Re-Im
          .iNumSegments   = {2465, 2467}  // Number of segments defined (needed with 'all segments' sweep)
        }
};
```

eg...
[```eChannel
getActiveChannelFrom8753learnString( guchar *pLearn, tGlobal *pGlobal)``` ](https://github.com/VK2BEA/HP8753-Companion/blob/6eabc218ea8f89b40a3acc6b5634a28c9c6572ef/src/HP8753comms.c#L1008)

[```gboolean
getStartStopOrCenterSpanFrom8753learnString( guchar *pLearn, tGlobal *pGlobal, eChannel channel ) ```](https://github.com/VK2BEA/HP8753-Companion/blob/6eabc218ea8f89b40a3acc6b5634a28c9c6572ef/src/HP8753comms.c#L1024C1-L1025C99)

also here
https://github.com/VK2BEA/HP8753-Companion/blob/6eabc218ea8f89b40a3acc6b5634a28c9c6572ef/src/HP8753comms.c#L1044

where I find the markers that are on and which is active, whether the analyzer is using Freq Start/Freq Stop or Center/Span. Also information on segments (not often used but when you need to look at several bands of frequencies without wasting points, its useful)
